zipkin-receiver-scribe/src/main/scala/com/twitter/zipkin/receiver/scribe/ScribeSpanReceiver.scala

```diff
@@ -111,11 +112,19 @@ class ScribeReceiver(
           batchesProcessedStat.add(spans.size)
           ok
         case Throw(NonFatal(e)) =>
-          if (!e.isInstanceOf[QueueFullException])
-            log.warning("Exception in process(): %s".format(e.getMessage))
-          errorStats.counter(e.getClass.getName).incr()
-          pushbackCounter.incr()
-          tryLater
+          // It is not an error if the scribe client decided to cancel its request.
+          // See Finagle FAQ's first entry http://twitter.github.io/finagle/guide/FAQ.html
+          if (e.isInstanceOf[CancellationException] && e.getCause.isInstanceOf[CancelledRequestException]) {
```

Comment:
I thought about that, but I think I'm out of scala.. is there a non-awkward way to look peek if the cause is CancelledRequestException using case?